### PR TITLE
More doctests and cleanup for intfuncs

### PR DESCRIPTION
**Pull request by kshyatt**

### CODE REVIEW

File: base/docs/helpdb/Base.jl [hunk 1]
```diff
@@ -1194,6 +1301,15 @@ cot
 
 Return the value stored for the given key, or the given default value if no mapping for the
 key is present.
+
+# Examples
+```jldoctest
+julia> get(d, "a", 3)
```
Comment (tkelman) [inline base/docs/helpdb/Base.jl]: does this pass? where does `d` come from?

Comment (kshyatt) [inline base/docs/helpdb/Base.jl]: whoops, forgot to copy paste

**Review by KristofferC — APPROVED** [review]
LGTM after one read through. It overlaps a bit with https://github.com/JuliaLang/julia/pull/22289 but no problems.

**Comment by KristofferC** [comment]
Think this should be valid to backport.

File: base/array.jl [hunk 2]
```diff
@@ -713,15 +748,14 @@ Resize `a` to contain `n` elements. If `n` is smaller than the current collectio
...
 ```jldoctest
 julia> resize!([6, 5, 4, 3, 2, 1], 3)
 3-element Array{Int64,1}:
  6
  5
  4
-```
 
-```julia-repl
 julia> resize!([6, 5, 4, 3, 2, 1], 8)
```
Comment (tkelman) [inline base/array.jl]: this isn't guaranteed to always be zero-initialized beyond the end of the array, is it?

Comment (fredrikekre) [inline base/array.jl]: No, I guess thats why this wasn't a doctest.

File: base/docs/helpdb/Base.jl [hunk 3]
```diff
@@ -1772,11 +1967,12 @@ Return the value stored for the given key, or if no mapping for the key is prese
 `key => f()`, and return `f()`.
 
 This is intended to be called using `do` block syntax:
-
+```julia
     get!(dict, key) do
```
Comment (tkelman) [inline base/docs/helpdb/Base.jl]: if fenced with triple backticks, then shouldn't be indented any more

File: base/docs/helpdb/Base.jl [hunk 4]
```diff
@@ -1821,6 +2018,19 @@ length(collection)
 
 Returns the index of the last value in `a` less than or equal to `x`, according to the
 specified order. Returns `0` if `x` is less than all values in `a`.
+
+# Examples
+
+```jldoctest
+julia> searchsortedlast([1, 2, 4, 5, 4], 4)
```
Comment (tkelman) [inline base/docs/helpdb/Base.jl]: examples for this function should use sorted inputs, that's what it's designed for

File: base/docs/helpdb/Base.jl [hunk 5]
```diff
@@ -2124,6 +2338,14 @@ throw
     ⊊(a,b) -> Bool
 
 Determine whether every element of `a` is also in `b`, using [`in`](@ref).
+
+```jldoctest
```
Comment (tkelman) [inline base/docs/helpdb/Base.jl]: `# Examples`

File: base/sparse/sparsematrix.jl [hunk 6]
```diff
@@ -2823,6 +2896,15 @@ stored and otherwise do nothing. Derivative forms:
 
     dropstored!(A::SparseMatrixCSC, i::Integer, J::AbstractVector{<:Integer})
     dropstored!(A::SparseMatrixCSC, I::AbstractVector{<:Integer}, j::Integer)
+
+# Example
+```jldoctest
+julia> Base.SparseArrays.dropstored!(A, [1, 2], [1, 1])
```
Comment (tkelman) [inline base/sparse/sparsematrix.jl]: where does A come from?

**Comment by kshyatt** [comment]
OK to go?

File: base/array.jl [hunk 7]
```diff
@@ -713,25 +748,27 @@ Resize `a` to contain `n` elements. If `n` is smaller than the current collectio
...
-8-element Array{Int64,1}:
- 6
- 5
- 4
- 3
- 2
- 1
- 0
- 0
+ julia> a = resize!([6, 5, 4, 3, 2, 1], 8);
```
Comment (tkelman) [inline base/array.jl]: looks like there's an extra leading space in this block

Comment (kshyatt) [inline base/array.jl]: Fixed, and I ran `make check-whitespace` locally.

Comment (tkelman) [inline base/array.jl]: how strange, I see bdbcbba from a fetch, but github isn't showing it in the pr yet?

File: base/docs/helpdb/Base.jl [hunk 8]
```diff
@@ -483,6 +499,23 @@ Returns the range of indices of `a` which compare as equal to `x` (using binary
 according to the order specified by the `by`, `lt` and `rev` keywords, assuming that `a`
 is already sorted in that order. Returns an empty range located at the insertion point
 if `a` does not contain values equal to `x`.
+
+# Examples
+
```
Comment (Sacha0) [inline base/docs/helpdb/Base.jl]: Extra empty line?

File: base/docs/helpdb/Base.jl [hunk 9]
```diff
@@ -786,6 +859,48 @@ at the position where it would appear if the array were fully sorted via a non-s
 algorithm. If `k` is a single index, that value is returned; if `k` is a range, an array of
 values at those indices is returned. Note that `select!` does not fully sort the input
 array.
+
+# Examples
+
```
Comment (Sacha0) [inline base/docs/helpdb/Base.jl]: Extra empty line here as well? (Have we a convention for absence/presence of an extra empty line following `# Example[s]`? Remarked here and above for consistency with the absence earlier in this PR, but have no preference.)

Comment (tkelman) [inline base/docs/helpdb/Base.jl]: we should pick and standardize - personally I lean towards leaving out the empty line, assuming it should render the same

File: base/docs/helpdb/Base.jl [hunk 10]
```diff
@@ -786,6 +859,48 @@ at the position where it would appear if the array were fully sorted via a non-s
...
+2
+
+julia> a
+5-element Array{Int64,1}:
+ 4
+ 4
+ 3
+ 2
+ 1
+```
```
Comment (Sacha0) [inline base/docs/helpdb/Base.jl]: Perhaps an example illustrating the partial sorting behavior would be worthwhile? The examples above yield fully sorted results.

File: base/docs/helpdb/Base.jl [hunk 11]
```diff
@@ -795,6 +910,15 @@ select!
 Create a random ASCII string of length `len`, consisting of upper- and
 lower-case letters and the digits 0-9. The optional `rng` argument
 specifies a random number generator, see [Random Numbers](@ref).
+
+# Example
+
```
Comment (Sacha0) [inline base/docs/helpdb/Base.jl]: See extra empty line question above :).

File: base/docs/helpdb/Base.jl [hunk 12]
```diff
@@ -1179,6 +1345,20 @@ delete!
 
 Returns the index of the first value in `a` greater than or equal to `x`, according to the
 specified order. Returns `length(a)+1` if `x` is greater than all values in `a`.
+`a` is assuemd to be sorted.
```
Comment (Sacha0) [inline base/docs/helpdb/Base.jl]: assuemd? :)

File: base/docs/helpdb/Base.jl [hunk 13]
```diff
@@ -1179,6 +1345,20 @@ delete!
...
+
+# Examples
+
```
Comment (Sacha0) [inline base/docs/helpdb/Base.jl]: See question above re. empty lines :).

File: base/docs/helpdb/Base.jl (hunk 12 above)
Comment (kshyatt) [inline base/docs/helpdb/Base.jl]: whoops

File: base/docs/helpdb/Base.jl (hunk 9 above)
Comment (kshyatt) [inline base/docs/helpdb/Base.jl]: I agree but I think we should a) pick one and b) have one mega-PR to fix it everywhere (much like @StefanKarpinski did with the `@test_approx_eq` macro).

File: base/multidimensional.jl [hunk 14]
```diff
@@ -111,6 +142,18 @@ module IteratorsMD
 
     Consequently these can be useful for writing algorithms that
     work in arbitrary dimensions.
+
```
Comment (Sacha0) [inline base/multidimensional.jl]: Perhaps an `# Example` heading?

**Review by Sacha0 — APPROVED** [review]
The new examples are a wonderful addition! :)

File: base/docs/helpdb/Base.jl (hunk 9 above)
Comment (Sacha0) [inline base/docs/helpdb/Base.jl]: Sounds good! :)

**Comment by kshyatt** [comment]
Unless someone requests changes or beats me, I'm going to merge in the morning :)

File: base/multidimensional.jl (hunk 14 above)
Comment (tkelman) [inline base/multidimensional.jl]: this wasn't addressed?

Comment (KristofferC) [inline base/multidimensional.jl]: https://github.com/JuliaLang/julia/pull/22565

**Pull request merged**